runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientFetchFactory.java

```diff
@@ -3098,13 +3124,8 @@ private void onDecodeFetchResponse(
 
                 if (topicPartitions.get(partitionId) == leaderId)
                 {
-                    fetchRequestCount++;
-                    if (fetchRequestCount == 2 && !dataFrameSent)
-                    {
-                        client.triggerFlushIfRequired(traceId);
-                    }
-                    dataFrameSent = false;
-                    signaler.signalNow(routeId, initialId, SIGNAL_NEXT_REQUEST, 0);
+                    doApplicationFlushIfNecessary(traceId, authorization, 0);
```

Comment:
If this is the only call site, and last argument is always `0`, then let's remove the last parameter from the method.